### DISCUSSION TITLE
GitHub Pages deployment is not working

### DISCUSSION THREAD

**Discussion by AlexK-1** [community/community#113276] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Today I created a new public repository and added the code of my site to it. It has `index.html` the file (in the root directory), and on the local device, the site opened normally in the browser. The only branch of the repository is called `main` But **I am unable to deploy my repository on GitHub Pages.** It says on the main repository page that one deployment was successfully performed. There are no errors in the logs. There is a 404 error on the page of the site I have deployed. There are no errors in the settings in the GitHub Pages section, but there is also no message about a successful deployment. I acted as it is written on the [GitHub Pages website](https://pages.github.com/). I tried clearing the browser cache. **What did I do wrong and how can I fix it?**

*If a link to the repository is needed, I am ready to provide it.*

**Comment by Blaise-93** [comment]
How to successfully deploy your website on GitHub page, ensure you follow the below steps:

1. Make sure you successful commit your work/projects on GitHub, using the following 3 git commands: a) git add . b) git commit -m <descriptive message log for each commit > and c) git pull and git push.
Now that your codes are on GitHub. Let's take it from there.

2. Go to your repo settings, in the "Code and automation" section of the sidebar, click Pages. Under "Build and deployment", under "Source", select Deploy from a branch.
 Under "Build and deployment", under "Branch", use the branch drop-down menu and select a publishing source. Optionally, open the README.md file of your repository

3. Use a deployment environment called `github-pages` from a starter workflow, and add it to your project folder root, do commit and push it as well. If your repository does already include an environment called github-pages , the environment will be created automatically.

I strongly recommend that you add a deployment protection rule so that only the default branch can deploy to this environment.

Make sure that all your file structure are well set, and if there is anything to configure, let's say you are using phyton (Django) or NodeJs for your project, make sure your settings are well configured prior to deployment.


Good luck.

**Comment by AlexK-1** [comment]
> 3\. Use a deployment environment called `github-pages` from a starter workflow, and add it to your project folder root, do commit and push it as well. If your repository does already include an environment called github-pages , the environment will be created automatically.

I don't fully understand this point. Do you mean this?

![image](https://github.com/community/community/assets/154962283/40a7a264-0a3b-47d8-aa15-486ec52df052)

**Comment by DeandrePetersen** [comment]
Make sure GitHub Pages is set up to show your site from the right place. Your site's main file, index.html, is directly in the main part of your project, which is called the main branch. Check this setup in the "Pages" area of your project settings. There, you can see if your site is coming from the main branch and the top level, or root, of your project.

**Reply by AlexK-1** [reply]
My `index.html ` the file is located in the root directory in the `main` branch. GitHub Pages are set to the `main` branch, the root directory.

**Comment by AlexK-1** [comment, marked as answer]
After I made changes to the repository using the `git push` command, the problem disappeared by itself.

**Comment by huzaifa-khan00** [comment]
Hi, Guys, I am facing the same issue with the deployment, and set everything right and pushed the local repo along with the whole project code, and branch set to main and everything is set in github pages, there is no error and the deployment link is not working.
